backend/src/test/java/gov/cdc/usds/simplereport/api/AuditLoggingFailuresTest.java

```diff
@@ -0,0 +1,142 @@
+import gov.cdc.usds.simplereport.db.repository.TestEventRepository;
+import gov.cdc.usds.simplereport.service.OrganizationService;
+import gov.cdc.usds.simplereport.service.TimeOfConsentService;
+import gov.cdc.usds.simplereport.test_util.TestUserIdentities;
+import java.util.List;
+import org.hibernate.HibernateException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+
+class AuditLoggingFailuresTest extends BaseApiTest {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AuditLoggingFailuresTest.class);
+  @Autowired private TestRestTemplate _restTemplate;
+  @Autowired private OrganizationService _orgService;
+
+  @MockBean private ApiAuditEventRepository _auditRepo;
+  @MockBean private TestEventRepository _testEventRepo;
+  @MockBean private TimeOfConsentService _consentService;
+  @Captor private ArgumentCaptor<ApiAuditEvent> _eventCaptor;
+
+  private Facility _base;
+  private Person _patient;
+  private PatientLink _link;
+
+  @BeforeEach
+  void init() {
+    TestUserIdentities.withStandardUser(
+        () -> {
+          Organization org = _orgService.getCurrentOrganization();
+          _base = _orgService.getFacilities(org).get(0);
+          _patient = _dataFactory.createFullPerson(org);
+          TestOrder order = _dataFactory.createTestOrder(_patient, _base);
+          _link = _dataFactory.createPatientLink(order);
+        });
+  }
+
+  @Test
+  void graphqlQuery_databaseError_eventLogged() {
+    when(_testEventRepo.save(any(TestEvent.class))).thenThrow(new IllegalArgumentException("ewww"));
+    useOrgUserAllFacilityAccess();
+    ObjectNode args =
+        patientArgs()
+            .put("deviceId", _base.getDefaultDeviceType().getInternalId().toString())
+            .put("result", "NEGATIVE");
+    runQuery("submit-test", args, "ewww");
+    verify(_auditRepo).save(_eventCaptor.capture());
+    ApiAuditEvent event = _eventCaptor.getValue();
+    assertEquals(List.of("addTestResult"), event.getGraphqlErrorPaths());
+  }
+
+  @Test
+  void graphqlQuery_auditFailure_noDataReturned() {
+    when(_auditRepo.save(_eventCaptor.capture()))
+        .thenThrow(new IllegalArgumentException("naughty naughty"));
+    useOrgUserAllFacilityAccess();
+    ObjectNode args = patientArgs().put("symptoms", "{}").put("noSymptoms", true);
+    assertThrows(NullPointerException.class, () -> runQuery("update-time-of-test", args));
```

Comment:
Not really following this one. Shouldn't it have thrown an IllegalArgumentException?

Reply:
No, it should have thrown some￼ custom exception indicating the bad response from the server. I need to file a bug about that. The underlying ￼issue here￼ is that raising an exception in the instrumentation causes a 400 error, which is probably not what we want, but I'm not sure what to do about it ￼